flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -258,15 +253,15 @@ public static TaskManagerServices fromConfiguration(
 			taskManagerServicesConfiguration.getTaskManagerAddress(),
 			dataPort);
 
-		// this call has to happen strictly after the network stack has been initialized
-		final MemoryManager memoryManager = createMemoryManager(taskManagerServicesConfiguration);
-
 		final BroadcastVariableManager broadcastVariableManager = new BroadcastVariableManager();
 
+		// this call has to happen strictly after the network stack has been initialized
```

Comment:
This looks like a stale comment.
Originally, it looks like to be introduced for network pre-allocating direct memory first (another comment, already deleted). Now memory manager does not pre-allocate anything anymore, so they should interfere with each other. Anyways the memory usage should all fit together at the end without overlapping otherwise it is a bug.
I added a hotfix commit to remove the confusing comments.